### ISSUE TITLE
Hoisting

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#454]
Found the problem, it seems swc doesn't handle var/function hoisting well. Here're two cases I found:

```js
// case one
var a = 2;
function foo() {
  try {
    var a = 1;
    console.log(a);
  } catch (err) {
    // ignored
  }
  console.log(a);
}
```

output:

```js
// case one
'use strict';
var a = bar;
function foo() {
    try {
        var a1 = 1;
        console.log(a1);
    } catch (err) {
    }
    console.log(a);
}
```

```js
// case two
function a() {}
function foo() {
  function b() {
    a();
  }
  function a() {}
}
```

output:

```js
'use strict';
function a() {
}
function foo() {
    function b() {
        a();
    }
    function a1() {
    }
}
```

This can happen a lot when compiling intermediate code generated by rollup in .esm.js. The only code that needs to be transpiled are the `import`s and `export`s, I'll try other workarounds if you think this is unnecessary to fix.

_Originally posted by @cyjake in https://github.com/swc-project/swc/issues/451#issuecomment-555310922_

**Comment by cyjake** [comment]
awesome, looking forward to v1.0.51 release

btw, is there anything I can help with swc? non-rust stuff perhaps, still struggling with the language 😭

**Comment by kdy1** [comment]
@cyjake I published v1.0.51.

Invoking swc with large codebase (and finding error) is really helpful to me.

**Comment by cyjake** [comment]
haha guess i'm doing exactly that right now
